### PR TITLE
Add is_clinical to table Customer

### PR DESCRIPTION
**Pull request by islean**
## Description

Tied to issue #2205 wherein we need a field to discern for how long data should be stored for a customer. This pr adds a boolean called "is_clinical" to the customer field.

### Added

- Boolean is_clincal to table Customer, and support for setting it

### How to prepare for test

- [x] Ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] Paxa the environment: `paxa`
- [x] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b add-clinical-field-to-customer -a
    ```

### How to test

- [x] Do an "cg add customer".
- [x] Connect to the database.
- [x] Redeploy master.

### Expected test outcome

- [x] Check that the Customer table has an is_clinical field upon deployment.
- [x] Check that the Customer table drops said column upon deployment of Master.
- [x] Verify that the add customer CLI command gives an option to set is_clinical and verify that it works.
- [x] Take a screenshot and attach or copy/paste the output.

## Review

- [x] Tests executed by IO
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Comment by islean** [comment]
Tests working as expected:

image
image

```
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
remote: Enumerating objects: 70, done.
remote: Counting objects: 100% (70/70), done.
remote: Compressing objects: 100% (22/22), done.
remote: Total 70 (delta 53), reused 65 (delta 48), pack-reused 0
Unpacking objects: 100% (70/70), done.
From github.com:Clinical-Genomics/cg
 * [new branch]      2056-remove-direct-imports-of-demultiplex-fixtures -> origin/2056-remove-direct-imports-of-demultiplex-fixtures
   6375410..2f62056  add-clinical-field-to-customer -> origin/add-clinical-field-to-customer
   209c245..72e162f  add-intengration-tests-for-demux-post-processing -> origin/add-intengration-tests-for-demux-post-processing
   c93637d..f9f0a90  taxprofiler_sample_id -> origin/taxprofiler_sample_id
Switched to branch 'add-clinical-field-to-customer'
Your branch is behind 'origin/add-clinical-field-to-customer' by 2 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)
Updating 6375410..2f62056
Fast-forward
 alembic/versions/68e54d17f4f3_add_is_clinical_to_customer.py | 5 +++++
 1 file changed, 5 insertions(+)
/home/isak.ohlsson
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade c3fdf3a8a5b3 -> 68e54d17f4f3, add is_clinical to customer
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/isak.ohlsson
done.
Log deploy... done.
cg, version 38.1.1
```
```
θ61° [isak.ohlsson@hasta:~] [S_base] 3m21s $ cg add customer --help
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))
Usage: cg add customer [OPTIONS] INTERNAL_ID NAME

  Add a new customer with a unique internal id and name.

Options:
  -cg, --collaboration-internal-ids TEXT
                                  List of internal IDs for the collaborations
                                  the customer should belong to
  -ia, --invoice-address TEXT     Street address, Post code, City  [required]
  -ir, --invoice-reference TEXT   Invoice reference (text)  [required]
  -da, --data-archive-location TEXT
                                  Specifies where to store data for the
                                  customer.  [default: PDC]
  -ic, --is-clinical              Flag stating if it is a clinical customer.
                                  [default: False]
  --help                          Show this message and exit.
```
```
θ61° [isak.ohlsson@hasta:~] [S_base] 22s $ cg add customer -ia "Somewhere in Bredäng", -ir "Best reference" -ic Cust1997 Isak
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Demux.samples to refer to viewonly relationship Sample.demuxes should include sync_backref=False set on the Demux.samples relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Sample.demuxes to refer to viewonly relationship Demux.samples should include sync_backref=False set on the Sample.demuxes relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
customer added: Cust1997 (177)
```

image

**Comment by islean** [comment]
Downgrade worked:

```
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))

INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running downgrade 68e54d17f4f3 -> c3fdf3a8a5b3, add is_clinical to customer
Switched to branch 'master'
Already up-to-date.
/home/isak.ohlsson
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/isak.ohlsson
done.
Log deploy... done.
cg, version 38.1.1
```
with no trace of the column anymore.

File: cg/cli/add.py [hunk 1]
```diff
@@ -62,6 +62,16 @@ def add():
     show_default=True,
     required=False,
 )
+@click.option(
+    "-ic",
+    "--is-clinical",
+    "is_clinical",
+    help="Flag stating if it is a clinical customer.",
```
Comment (diitaz93) [inline cg/cli/add.py:69]: Maybe for the users of this function may be obvious, but a short explanation of what is a clinical customer would be better, as the flag is self-explanatory about the status but not about the meaning.

**Review by diitaz93 — APPROVED** [review]
It looks good, just wondering if the help message could be improved 🙂

File: cg/cli/add.py (hunk 1 above)
Comment (islean) [reply cg/cli/add.py:69]: Reasonable, will add that this will be used to determine for how long the data will be stored.

**Comment by islean** [comment]
Retested since the production alembic revision has changed. Same nice result.

File: alembic/versions/68e54d17f4f3_add_is_clinical_to_customer.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+"""add is_clinical to customer
+
+Revision ID: 68e54d17f4f3
+Revises: c3fdf3a8a5b3
```
Comment (ChrOertlin) [inline alembic/versions/68e54d17f4f3_add_is_clinical_to_customer.py:4]:
```suggestion
Revises: 2fdb42ba801a
```

**Review by ChrOertlin — APPROVED** [review]
just the docstring with current version

**Pull request merged**